src/main/java/org/owfs/jowfsclient/internal/OwfsConnectionImpl.java

```diff
@@ -74,16 +75,19 @@ private boolean connect(boolean force) throws IOException {
 	}
 
 	private boolean tryToSocketConnectionAndStreamsInitialization() throws IOException {
-		owSocket = new Socket();
+	    owSocket = new Socket();
+	    for (InetAddress inetAddress : InetAddress.getAllByName(config.getHostName())) {
 		try {
-			owSocket.connect(new InetSocketAddress(config.getHostName(), config.getPortNumber()), config.getConnectionTimeout());
-			owIn = new DataInputStream(owSocket.getInputStream());
-			owOut = new DataOutputStream(owSocket.getOutputStream());
-			return true;
```

Comment:
What if the first inet address is an ipv6 adress, we would end up in the same situation, no?